src/parser.ts

```diff
@@ -701,60 +712,73 @@ export class Parser {
+        let defaultValue: { value: any } | null = null;
 
-      let defaultValue: { value: any } | null = null;
+        if (hasCodeBasedDefault) {
+          defaultValue = { value: defaultProps[propName] };
+        } else if (jsDocComment.tags.default) {
+          defaultValue = { value: jsDocComment.tags.default };
+        }
 
-      if (hasCodeBasedDefault) {
-        defaultValue = { value: defaultProps[propName] };
-      } else if (jsDocComment.tags.default) {
-        defaultValue = { value: jsDocComment.tags.default };
-      }
+        const parents = getDeclarations(prop);
+        const declarations = prop.declarations || [];
+        const baseProp = baseProps.find(p => p.getName() === propName);
 
-      const parent = getParentType(prop);
-      const parents = getDeclarations(prop);
-      const declarations = prop.declarations || [];
-      const baseProp = baseProps.find(p => p.getName() === propName);
-
-      const required =
-        !isOptional(prop) &&
-        !hasCodeBasedDefault &&
-        // If in a intersection or union check original declaration for "?"
-        // @ts-ignore
-        declarations.every(d => !d.questionToken) &&
-        (!baseProp || !isOptional(baseProp));
-
-      const type = jsDocComment.tags.type
-        ? {
-            name: jsDocComment.tags.type
-          }
-        : this.getDocgenType(propType, required);
-
-      const propTags = this.shouldIncludePropTagMap
-        ? { tags: jsDocComment.tags }
-        : {};
-      const description = this.shouldIncludePropTagMap
-        ? jsDocComment.description.replace(/\r\n/g, '\n')
-        : jsDocComment.fullComment.replace(/\r\n/g, '\n');
-
-      result[propName] = {
-        defaultValue,
-        description: description,
-        name: propName,
-        parent,
-        declarations: parents,
-        required,
-        type,
-        ...propTags
-      };
+        const required =
+          !isOptional(prop) &&
+          !hasCodeBasedDefault &&
+          // If in a intersection or union check original declaration for "?"
+          // @ts-ignore
+          declarations.every(d => !d.questionToken) &&
+          (!baseProp || !isOptional(baseProp));
+
+        const type = jsDocComment.tags.type
+          ? {
+              name: jsDocComment.tags.type
+            }
+          : this.getDocgenType(propType, required);
+
+        const propTags = this.shouldIncludePropTagMap
+          ? { tags: jsDocComment.tags }
+          : {};
+        const description = this.shouldIncludePropTagMap
+          ? jsDocComment.description.replace(/\r\n/g, '\n')
+          : jsDocComment.fullComment.replace(/\r\n/g, '\n');
+
+        const propItem: PropItem = {
+          defaultValue,
+          description: description,
+          name: propName,
+          parent,
+          declarations: parents,
+          required,
+          type,
+          ...propTags
+        };
+        if (parent?.fileName.includes('node_modules')) {
```

Comment:
I put it like this at first, because I saw a ton of types coming from `@types/react` for all the native props. But I guess we could also cache user defined types in here as well.